src/lib/updateGrantsRecipients.js

```diff
@@ -400,8 +400,13 @@ export async function processFiles(hashSumHex) {
 
       await updateCDIGrantsWithOldGrantData(cdiGrantsToLink);
 
+      // Deduplicate based on 'id'
+      const uniqueProgramsForDb = Array.from(
+        new Map(programsForDb.map((item) => [item.id, item])).values(),
```

Comment:
In this case, given a program type array like:
```js
[{ id: 1, name: "EHS"}, { id: 1, name: "HS"}, { id: 2, name: "HS"}, { id: 3, name: "HS"}]
```

This code will return 
```js
[{ id: 1, name: "HS"}, { id: 2, name: "HS"}, { id: 3, name: "HS"}]
```
meaning the last item with that id in the array is the one recorded. 

Is there any chance of those values diverging or are the duplicates returned from HSES always identical?

Reply:
We use their ids that need to be unique, so there shouldn't be any different records with the same `id`.  Good thought though.

The current duplicates are identical with respect to the data that we extract. The full HSES records only differ in their representation of null values for certain address fields. My guess is that's the reason they are even included in the feed. HSES is looking into it, so hopefully, it will be fixed on their end soon as well.

Reply:
Having the duplicate records blocks our data updates though.